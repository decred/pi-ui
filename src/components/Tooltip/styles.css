.tooltipContent {
  border-radius: 0.2rem;
  box-shadow: 0.1rem 0.1rem 0.6rem 0.25rem var(--color-shadow-light);
  padding: 0.25rem 1.5rem;
  user-select: none;
  max-width: 30rem;
  word-wrap: break-word;
  pointer-events: none;
  z-index: var(--z-index-big);
  background-color: var(--tooltip-background-color);
  visibility: visible;
  opacity: 0;
  transition: 0.3s all ease;
}

.tooltip:hover .tooltipContent {
  visibility: visible;
  opacity: 1;
}

.tooltip {
  position: relative;
  display: inline-block;
}

.top {
  position: absolute;
  margin: 0.6rem 0;
  bottom: 100%;
  left: 50%;
  transform: translateX(-50%);
}

.bottom {
  position: absolute;
  margin: 0.6rem 0;
  top: 100%;
  left: 50%;
  transform: translateX(-50%);
}

.left {
  position: absolute;
  margin: 0 0.6rem;
  right: 100%;
  top: 50%;
  transform: translateY(-50%);
}

.right {
  position: absolute;
  margin: 0 0.6rem;
  left: 100%;
  top: 50%;
  transform: translateY(-50%);
}
