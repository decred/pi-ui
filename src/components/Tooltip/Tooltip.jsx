import React, { useState } from "react";
import PropTypes from "prop-types";
import styles from "./styles.module.css";
import { classNames } from "../../utils";
import useMediaQuery from "../../hooks/useMediaQuery";
import useClickOutside from "../../hooks/useClickOutside";

const noop = () => ({});

const Tooltip = ({
  children,
  content,
  placement,
  className,
  contentClassName,
  ...props
}) => {
  // Uses 1024px screen width to include big iPad sizes
  const isMobile = useMediaQuery("(max-width: 1024px)");
  const [isActive, toggleIsActive] = useState(false);
  const showTooltip = () => toggleIsActive(true);
  const hideTooltip = () => toggleIsActive(false);
  const [tooltipRef] = useClickOutside(hideTooltip);
  return (
    <div
      ref={tooltipRef}
      className={classNames(styles.tooltip, className)}
      onClick={isMobile ? showTooltip : noop}
      {...props}>
      <div
        className={classNames(
          styles.tooltipContent,
          styles[placement],
          contentClassName,
          isActive && styles.showTooltip
        )}>
        {content}
      </div>
      {children}
    </div>
  );
};

Tooltip.propTypes = {
  children: PropTypes.node,
  content: PropTypes.node,
  placement: PropTypes.string,
  className: PropTypes.string,
  contentClassName: PropTypes.string,
  style: PropTypes.object,
};

Tooltip.defaultProps = {
  placement: "top",
  content: "content",
};

export default Tooltip;
