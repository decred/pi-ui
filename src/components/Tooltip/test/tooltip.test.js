import React from "react";
import Tooltip from "../Tooltip";
import { create } from "react-test-renderer";

describe("Tooltip component", () => {
  test("Matches the snapshot", () => {
    const tooltip = create(<Tooltip />);
    expect(tooltip.toJSON()).toMatchSnapshot();
  });
});
