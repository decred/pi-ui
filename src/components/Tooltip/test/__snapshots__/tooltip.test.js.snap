// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component Matches the snapshot 1`] = `
<div
  className="tooltip"
  onClick={[Function]}
>
  <div
    className="tooltipContent top"
  >
    content
  </div>
</div>
`;
